include/parakeet/ProE/internal/Parser.h

```diff
@@ -0,0 +1,139 @@
+/*
+	Copyright 2021 OpenJAUS, LLC (dba MechaSpin). Subject to the MIT license.
+*/
+
+#ifndef PARAKEET_PROE_PARSER_H
+#define PARAKEET_PROE_PARSER_H
+
+#include <functional>
+#include <memory>
+#include <vector>
+#include <chrono>
+
+namespace mechaspin
+{
+namespace parakeet
+{
+namespace ProE
+{
+namespace internal
+{
+class Parser
+{
+	public:
+		struct LidarSensorProperties
+		{
+			bool unitIsInCM;
+			bool withIntensity;
+			bool doDragPointRemoval;
+			bool doDataSmoothing;
+
+			uint32_t value;
+		};
+
+		struct LidarPoint
+		{
+			uint16_t distance;
+			uint16_t relativeStartAngle;
+			uint8_t intensity;
+		};
+
+		struct CompleteLidarMessage
+		{
+			uint16_t totalPoints;
+			uint32_t startAngle;
+			uint32_t endAngle;
+
+			LidarSensorProperties sensorPropertyFlags;
+
+			std::chrono::system_clock::time_point timestamp;
+			uint32_t deviceNumber;
+
+			std::vector<LidarPoint> lidarPoints;
+		};
+
+		Parser(std::function<void(CompleteLidarMessage*)> onCompleteLidarMessageCallback);
+
+		int parseSensorBuffer(int length, unsigned char* buf);
+
+		void reset();
+
+	private:
+		struct BufferData
```

Comment:
Yeah, I would probably move this out to be a public standalone struct (likely internal to start) and pass around the struct vs the char* and length up until the end.

Reply:
Agree this is the way forward, my question is where it should be located. Keeping it within UdpSocket / Parser does not make sense as they would need to include each other. Should I make a separate file which has these structs? If so, what should it be called? IPv4Structs?